src/main/java/com/oath/halodb/HaloDB.java

```diff
@@ -40,6 +40,10 @@ public static HaloDB open(String directory, HaloDBOptions opts) throws HaloDBExc
         }
     }
 
+    public int size(byte[] key) {
```

Comment:
Is 'size' the right name?  there is already a `size` method that returns the number of elements.   To avoid confusion, we should consider a different name.  Perhaps `entrySize` or `valueSize` ?

Also, the public API in HaloDB.java should document the expected behavior -- something like `Returns the size of the value if present, or -1 if not present.`